internal/sourceanalysis/go.go

```diff
@@ -39,14 +50,18 @@ func matchAnalysisWithPackageVulns(pkgs []models.PackageVulns, gvcResByVulnID ma
 					*analysis = make(map[string]models.AnalysisInfo)
 				}
 
-				gvcVuln, ok := gvcResByVulnID[vulnID]
+				moduleToCalled, ok := idToModuleToCalled[vulnID]
 				if !ok { // If vulnerability not found, check if it contain any source information
 					fillNotImportedAnalysisInfo(vulnsByID, vulnID, pv, analysis)
 					continue
 				}
 				// Module list is unlikely to be very big, linear search is fine
-				containsModule := slices.ContainsFunc(gvcVuln.Modules, func(module *govulncheck.Module) bool {
-					return module.Path == pv.Package.Name
+				var modulePaths []string
```

Comment:
This part can be replaced a direct map key check, something like:
```
called, hasModule := moduleToCalled[pv.Package.Name];
(*analysis)[vulnID] = models.AnalysisInfo{
	Called: hasModule && called,
}
```

Reply:
Nice catch! I left out the `hasModule` bit, since we want to set `Called` to `false` regardless of if `pv.Package.Name` is not in the map, or is present but has a value of `false`